app/persistence/status/status.go

```diff
@@ -1,5 +1,63 @@
 package status
 
+import (
+	"errors"
+	"github.com/hashgraph/hedera-sdk-go"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+	"strconv"
+	"time"
+)
+
 // This table will contain information for latest status of the application
 type Status struct {
+	Name      string
+	AccountID string
+	Code      string
+	Value     string
+}
+
+type StatusRepository struct {
+	dbClient *gorm.DB
+}
+
+func NewStatusRepository(dbClient *gorm.DB) *StatusRepository {
+	return &StatusRepository{
+		dbClient: dbClient,
+	}
+}
+
+func (s StatusRepository) GetLastFetchedTimestamp(accountID hedera.AccountID) string {
+	lastFetchedStatus := &Status{}
+	failure := s.dbClient.
+		Table("statuses").
+		Where("code = ? and account_id = ?", "LAST_FETCHED_TIMESTAMP", accountID.String()).
+		First(&lastFetchedStatus).Error
+
+	if failure != nil && errors.Is(failure, gorm.ErrRecordNotFound) {
+		log.Errorf("Could not get last fetched timestamp: [%s]\n", failure)
+		now := time.Now()
```

Comment:
We shouldn't be setting "last fetched timestamp" before we actually fetched it. You should return `nil` or error and handle that in the watcher.